### PR TITLE
Refine Clipboard Suggestion Validation to Prevent Empty and Invalid Items

### LINKED ISSUE #2386: Incorrect Suggestions from Empty or Invalid Clipboard Content
### Short description

- When external apps like KDE Connect update the clipboard with empty content or content containing only special characters (newlines, tabs), the keyboard provides inaccurate or unnecessary suggestions.
- This can lead to a confusing and potentially frustrating user experience.

### Steps to reproduce

1. Install and configure an app like KDE Connect that can share clipboard content.
2. Use KDE Connect to share an empty file or text containing only newlines or tabs.
3. Observe the application displaying suggestions for the empty clipboard content.

### Expected Behavior:
- The keyboard should not generate suggestions for empty or invalid clipboard content.

### Actual Behavior:
- The keyboard generates suggestions for the empty clipboard content.

### FlorisBoard Version

0.4.0-alpha06

### Install Source

GitHub

### Device

Pixel 4a

### Android

Android 13

### PR DESCRIPTION
**Pull request by snaik20**
### Summary:
- This PR addresses an issue where empty or invalid clipboard items that could lead to incorrect suggestions.
- It introduces validation logic to ensure only meaningful content is suggested.

### Context:
- The issue was observed when apps like KDE Connect added empty or special character-only items to the clipboard.
- The `ClipboardSuggestionProvider` previously lacked validation for such cases.

### Changes:
- Added validation for empty clipboard items.
- Enhanced validation to handle special characters (newlines, tabs).
- Introduced private methods and constants for improved readability and maintainability.

Fixes #2386

### CODE REVIEW

File: app/src/main/kotlin/dev/patrickgold/florisboard/ime/nlp/NlpManager.kt [hunk 1]
```diff
@@ -55,6 +56,10 @@ import java.util.concurrent.atomic.AtomicBoolean
 import java.util.concurrent.atomic.AtomicInteger
 import kotlin.properties.Delegates
 
+private const val NEW_LINE = "\n"
+private const val TAB = "\t"
+private const val SPACE = ""
```
Comment (patrickgold) [inline app/src/main/kotlin/dev/patrickgold/florisboard/ime/nlp/NlpManager.kt:61]:
```suggestion
private const val BLANK_STR_REGEX = Regex("^\\s*$")
```

File: app/src/main/kotlin/dev/patrickgold/florisboard/ime/nlp/NlpManager.kt [hunk 2]
```diff
@@ -459,5 +462,15 @@ class NlpManager(context: Context) {
...
         }
+
+        private fun validateClipboardItem(currentItem: ClipboardItem?, lastItemId: Long, contentText: String) =
+            currentItem?.takeIf {
+                // Check if already used
+                it.id != lastItemId
+                    // Check if content is empty
+                    && contentText.isBlank()
+                    // Check if clipboard content has any valid characters
+                    && currentItem.text?.replace(NEW_LINE, SPACE)?.replace(TAB, SPACE)?.isBlank() == false
```
Comment (patrickgold) [inline app/src/main/kotlin/dev/patrickgold/florisboard/ime/nlp/NlpManager.kt:473]:
```suggestion
                    && !BLANK_STR_REGEX.matches(currentItem.text ?: "")
```

**Review by patrickgold — CHANGES_REQUESTED** [review]
Thanks a lot for your changes!

I have reviewed your code and in general like your idea. However there's a better and more streamlined version to do blank str matching using regex, see the suggested changes. `\s` matches quite a few blank characters and avoids us hard-coding what a blank character is, see the [Java docs for Pattern](https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html)

File: app/src/main/kotlin/dev/patrickgold/florisboard/ime/nlp/NlpManager.kt (hunk 1 above)
Comment (snaik20) [reply app/src/main/kotlin/dev/patrickgold/florisboard/ime/nlp/NlpManager.kt]: Done.

File: app/src/main/kotlin/dev/patrickgold/florisboard/ime/nlp/NlpManager.kt (hunk 2 above)
Comment (snaik20) [reply app/src/main/kotlin/dev/patrickgold/florisboard/ime/nlp/NlpManager.kt]: Done

**Comment by snaik20** [comment]
> Thanks a lot for your changes!
> 
> I have reviewed your code and in general like your idea. However there's a better and more streamlined version to do blank str matching using regex, see the suggested changes. `\s` matches quite a few blank characters and avoids us hard-coding what a blank character is, see the [Java docs for Pattern](https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html)

Sounds great. Verified the new changes and updated PR.

**Review by patrickgold — APPROVED** [review]
Thanks a lot for your changes, now everything looks good, merging in!

**Pull request merged**